src/lib_json/json_reader.cpp

```diff
@@ -18,6 +18,8 @@
 #include <memory>
 #include <set>
 #include <limits>
+#include <stdio.h>
```

Comment:
Ahh sorry I was mistaken. It wasn't VS2010, it was a ghetto version of ARMCC I have to use. Each of the files I added stdio.h or string.h to already had `<cstring>` and` <cstdio>` in it, and it wasn't sufficient--so your test is already done.

I noticed this snippet in one place (and snippets like it in many other places)

> #if defined(`__QNXNTO__`)
> #define sscanf std::sscanf
> #endif`

This is representative of another solution to the actual problem I had (symbols like sscanf being undeclared, being in the std:: namespace instead). So, it seems you've dealt with this problem before. 

Since I don't know anything about compilers from the present or the future (it's all the same to a guy stuck in the past) all I can do is naively google up things like http://en.cppreference.com/w/cpp/io/c/sscanf which seems to say those functions belong in std:: namespace indeed and assume your great modern compilers which support them without the std:: qualifier are leading you astray

I'll let you think on this a while before I do any reverting. I can't competently suggest anything, but can incompetently suggest you sprinkle the code with std:: liberally if you want to be standards-compliant with c++29 or whatever you're targeting. For instance, I fixed json_writer.cpp in a different way by adding six std::